modules/bedtools_bamToBed.cfmod.pl

```diff
@@ -0,0 +1,127 @@
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+	'cores' 	=> ['2', '4'],
+	'memory' 	=> ['8G', '16G'],
+	'modules' 	=> 'BEDTools',
+	'time' 		=> sub {
+
+		my $cf = $_[0];
+		my $num_files = $cf->{'num_starting_merged_aligned_files'};
+		# Default to 1 if none were parsed for whatever reason
+		$num_files = ($num_files > 0) ? $num_files : 1;
+		# To be safe, we set the time limit to 1 hour per bam file.
+		return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 60);
+	}
+);
+
+# The help text
+my $helptext = "".("-"x15)."\n Bedtools bamToBed\n".("-"x15)."\n
+Converts bam files to bed format and sorts the bed file, using bedtools.
+Outputs are sorted bed files named basename.bed.\n\n";
+
+# Start your engines...
+my %cf = CF::Helpers::module_start(\%requirements, $helptext);
+
+# Print version information about the program to be executed.
+warn "---------- Bedtools version information ----------\n";
+warn `bedtools --version`;
+warn "\n------- End of bedtools version information ------\n";
+
+# Open up our run file in append mode
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Go through each file and run our command
+foreach my $file (@{$cf{'prev_job_files'}}){
+  # Start the clock...
+  my $timestart = time;
+
+  # Check that input is bed file
+  if(!($file =~ /\.bam$/)){
+      warn "\n###CF Error! bedtools_bamToBed failed: bam file expected, got $file\n\n";
+      next;
+  }
+  
+  # Generate a nice output file name
+  my $output_fn = $file.".bed";
+  my $tmpFile = $file."_TOSORT.bed";
+
+
+  # Convert to bed format
+  my $cmd = "bamToBed -i $file > $tmpFile";
+  warn "\n###CFCMD $cmd\n\n";
+  
+  # Try to run the command - returns 0 on success (which evaluated to false)
+  if(!system ($cmd)){
+    # Command worked!
+    # Work out how long the processing took
+    my $duration =  CF::Helpers::parse_seconds(time - $timestart);
+    
+    # Print a success message to the log file which will be e-mailed out
+    warn "###CF bamToBed successfully exited, took $duration..\n";
+  } else {
+    # Command returned a non-zero result, probably went wrong...
+    warn "\n###CF Error! Bedtools bamToBed failed: $? $!\n\n";
```

Comment:
Thanks! I’ll go these changes soon, but I already forgot what the simplest
way to incorporate the though git was..

Should I add them to my version on uppmax, then push to gut hub, then do
another pull request?

/Jakub

On Tuesday, June 23, 2015, Phil Ewels <notifications@github.com> wrote:
> In modules/bedtools_bamToBed.cfmod.pl:
>
>> +
>> + # Convert to bed format
>> + my $cmd = "bamToBed -i $file > $tmpFile";
>> + warn "\n###CFCMD $cmd\n\n";
>> +
>> + # Try to run the command - returns 0 on success (which evaluated to
false)
>> + if(!system ($cmd)){
>> + # Command worked!
>> + # Work out how long the processing took
>> + my $duration = CF::Helpers::parse_seconds(time - $timestart);
>> +
>> + # Print a success message to the log file which will be e-mailed out
>> + warn "###CF bamToBed successfully exited, took $duration..\n";
>> + } else {
>> + # Command returned a non-zero result, probably went wrong...
>> + warn "\n###CF Error! Bedtools bamToBed failed: $? $!\n\n";
>
> Should we add a next here to avoid trying to sort the file after the
conversion failed?
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci4.googleusercontent.com/proxy/USw0aPYKlP0Qk4O1yauIDAd-MON1BkVrykU6_F6qdNE8fx87VvQ0v6tNUg1JSO3GxNLGrUSyaBDeFjmtEVXShztIr4vf065-VoUYE24Om8KfiJSpff2VsSWui14O5_zR0O-tCoh_B1p677AyvxcccsvIbB3nng=s0-d-e1-ft#https://github.com/notifications/beacon/AH70iXyPwn64_eyMxvX5WBHGd-Nh8V9vks5oWVb0gaJpZM4FI3GN.gif
>